lib/projects.rb

```diff
@@ -0,0 +1,33 @@
+require_relative "project_printer"
+
+class Projects
+  attr_reader :projects, :project_printer 
+  
+  def initialize(project_printer)
+    @projects = []
+    @project_printer = project_printer
+  end
+
+  def projects_empty?
+    projects.empty?
+  end
+
+  def add_project(name)
+    projects << {name => []}
+  end
+
+  def current_project_exists?(name)
+    projects.detect { |p| p.keys.first == name }
+  end
+
+  def list
```

Comment:
listing is about printing and not about the data. You need to iterate through each item is the data, but that `each` or even `map` is the thing to expose here. So you would want something that takes a block and eaches or maps over the elements passing your project into the block.